player/stream.go

```diff
@@ -192,8 +199,11 @@ func (s *Stream) GetChunk(chunkIdx int) (ReadableChunk, error) {
 		return nil, err
 	}
 
-	if s.player.prefetch {
-		go s.prefetchChunk(chunkIdx + 1)
+	chunkToPrefetch := chunkIdx + 1
+	prefetched := s.prefetchedChunks[chunkToPrefetch]
+	if s.player.prefetch && !prefetched {
+		s.prefetchedChunks[chunkToPrefetch] = true
```

Comment:
is this threadsafe? why doesn't this panic in prod?

Reply:
each stream has its own allocated structure so it's thread safe as in it doesn't interact with other people/tabs streaming the same video.

I don't think this region of code is ever concurrently accessed